### PR TITLE
CLOUDP-167803: Migrate network peering connections

### PR DESCRIPTION
**Pull request by andmatei**
## Proposed changes
Migrate network peering connections store to v2 client.

_Jira ticket:_ CLOUDP-167803

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/store/peering_connections.go [hunk 1]
```diff
@@ -25,16 +25,16 @@ import (
...
+	PeeringConnection(string, string) (interface{}, error)
 }
 
 type PeeringConnectionCreator interface {
 	CreateContainer(string, *atlasv2.CloudProviderContainer) (interface{}, error)
-	CreatePeeringConnection(string, *atlas.Peer) (*atlas.Peer, error)
+	CreatePeeringConnection(string, *atlasv2.ContainerPeerViewRequest) (interface{}, error)
 }
 
 type AzurePeeringConnectionCreator interface {
```
Comment (andmatei) [inline internal/store/peering_connections.go:40]: We can consider to add specific methods for retrieving and creating networking peering connections based on the cloud provider, i.e., these interfaces can specific methods like `CreateAzurePeeringConnection`

File: internal/cli/atlas/networking/peering/watch.go [hunk 2]
```diff
@@ -56,10 +57,28 @@ func (opts *WatchOpts) watcher() (bool, error) {
...
+		return watcherAzure(v), nil
+	case *atlasv2.GCPPeerVpc:
+		return watcherGCP(v), nil
 	}
-	return result.StatusName == pendingAcceptance || result.StatusName == failed || result.StatusName == available, nil
+	return false, nil
+}
+
+func watcherGCP(peer *atlasv2.GCPPeerVpc) bool {
+	return *peer.Status == waitingForUser || *peer.Status == failed || *peer.Status == available
```
Comment (gssbzn) [inline internal/cli/atlas/networking/peering/watch.go:73]: given these are pointers I believe we generate getters if you have a `GetStatus()` method you can simplify this a lot 
```go
type StatusGetter interface {
  GetStatus()(something)
}

func watcher(peer StatusGetter) bool {
  return peer.GetStatus() == waitingForUser || peer..GetStatus() == failed || peer..GetStatus() == available
}
```
by using an interface you can make it a single method

Comment (andmatei) [reply internal/cli/atlas/networking/peering/watch.go:73]: Great suggestion! The only problem for this is that the AWS method is called `GetStatusName()`, and the GCP and Azure ones are called `GetStatus()` - and I don't know if there is any Golang magic that we could use in this case

File: test/e2e/atlas/kubernetes_config_generate_test.go [hunk 3]
```diff
@@ -638,10 +639,10 @@ func TestProjectWithNetworkPeering(t *testing.T) {
 		t.Cleanup(func() {
 			deleteAllNetworkPeers(t, cliPath, generator.projectID, gcpEntity)
 		})
-		var createdNetworkPeer mongodbatlas.Peer
+		var createdNetworkPeer mongodbatlasv2.GCPPeerVpc
 		err = json.Unmarshal(resp, &createdNetworkPeer)
 		require.NoError(t, err)
-		expectedProject.Spec.NetworkPeers[0].ContainerID = createdNetworkPeer.ContainerID
+		expectedProject.Spec.NetworkPeers[0].ContainerID = createdNetworkPeer.ContainerId
```
Comment (matt-condon) [inline test/e2e/atlas/kubernetes_config_generate_test.go:645]: Can be a separate PR, but would it be worthwhile or overkill to have e2e tests for each of GCP/Azure/AWS @gssbzn ?

**Review by matt-condon — APPROVED** [review]
LGTM nice work!

**Review by gssbzn — APPROVED** [review]
LGTM

**Review by wtrocki — APPROVED** [review]
Reviewed changes. No verification done.
